test/dlc_tests/conftest.py

```diff
@@ -222,10 +235,111 @@ def pytest_runtest_setup(item):
+            github_link = f"https://github.com/aws/deep-learning-containers/blob/master/" \
+                          f"{str_fspath.split('/deep-learning-containers/')[-1]}"
+
+            # Create a new test coverage item if we have not seen the function before. This is a necessary step,
+            # as parametrization can make it appear as if the same test function is a unique test function
+            if not test_cov.get(function_key):
+                test_cov[function_key] = {
+                                            "Category": category,
+                                            "Name": function_name,
+                                            "Scope": scope,
+                                            "Job_Type": train_inf,
+                                            "Num_Instances": get_marker_arg_value(item, "multinode", num_instances),
+                                            "Processor": cpu_gpu,
+                                            "Integration": get_marker_arg_value(item, "integration", integration),
+                                            "Model": get_marker_arg_value(item, "model", model),
+                                            "Github_Link": github_link,
+                                           }
+        with open("test_coverage_report.csv", "w+") as tmp_file:
+            # Assemble the list of headers from one item in the dictionary
+            field_names = []
+            for _key, header in test_cov.items():
+                for field_name, _value in header.items():
+                    field_names.append(field_name)
+                break
+
+            writer = csv.DictWriter(tmp_file, delimiter=",", fieldnames=field_names)
+            writer.writeheader()
+
+            for _func_key, info in test_cov.items():
+                writer.writerow(info)
+
+
+class RequiredMarkerNotFound(Exception):
+    pass
+
+
+def infer_field_value(default, options, *comparison_str):
+    """
+    For a given test coverage report field, determine the value based on whether the options are in keywords or
+    file paths.
+
+    :param default: default return value if the field is not found
+    :param options: tuple of possible options -- i.e. ("training", "inference")
+    :param comparison_str: keyword string, filepath string
+    :return: field value <str>
+    """
+    for comp in comparison_str:
+        for option in options:
+            if option in comp:
+                return option.strip("_")
+    return default
+
+
+def get_marker_arg_value(item_obj, marker_name, default=None):
+    """
+    Function to return the argument value of a pytest marker -- if it does not exist, fall back to a default.
+    If the default does not exist and the option does not exist, raise an error.
+
+    :param item_obj: pytest item object
+    :param marker_name: name of the pytest marker
+    :param default: default return value -- if none,
```

Comment:
nit: fill in the rest of this docstring. If None, assume this is a required marker